apple/internal/outputs.bzl

```diff
@@ -40,15 +40,21 @@ def _archive(
         bundle_name,
         executable_name = None,
         platform_prerequisites,
-        predeclared_outputs):
+        predeclared_outputs,
+        rule_descriptor,
+        label_name):
     """Returns a file reference for this target's archive."""
     bundle_name_with_extension = bundle_name + bundle_extension
 
     tree_artifact_enabled = is_experimental_tree_artifact_enabled(
         config_vars = platform_prerequisites.config_vars,
     )
     if tree_artifact_enabled:
-        return actions.declare_directory(bundle_name_with_extension)
+        archive_relative_path = rule_descriptor.bundle_locations.archive_relative
+        root_path = label_name + "_archive-root"
```

Comment:
Why `_archive-root`?

Also, can you add a test that shows this new non-clashing behavior?

Reply:
I initially started without the `_archive-root` but then the builds failed saying something related to common prefixes.
```
bazel build //examples/ios/HelloWorld:HelloWorld  //examples/ios/HelloWorld:HelloWorld_Beta --define=apple.experimental.tree_artifact_outputs=1 
ERROR: One of the output paths 'bazel-out/applebin_ios-ios_sim_arm64-fastbuild-ST-81e85097ebcb/bin/examples/ios/HelloWorld/HelloWorld_Beta' (belonging to //examples/ios/HelloWorld:HelloWorld_Beta) and 'bazel-out/applebin_ios-ios_sim_arm64-fastbuild-ST-81e85097ebcb/bin/examples/ios/HelloWorld/HelloWorld_Beta/Payload/HelloWorld.app' (belonging to //examples/ios/HelloWorld:HelloWorld_Beta) is a prefix of the other. These actions cannot be simultaneously present; please rename one of the output files or build just one of them
```

Reply:
Added the test 👍🏼